library/asn1parse.c

```diff
@@ -420,6 +429,10 @@ int mbedtls_asn1_get_alg(unsigned char **p,
     params->tag = **p;
     (*p)++;
 
+    if (mbedtls_asn1_bad_tag(params->tag)) {
```

Comment:
This is only useful if the caller doesn't subsequently check the tag, which seems unlikely. If the caller doesn't check the tag, then with this patch we'll reject a guaranteed-invalid tag but not a potentially valid tag that is not understood, so the benefit is very limited.